common/OpTestUtil.py

```diff
@@ -2353,6 +2353,123 @@ def prepare_source(self,spec_file, host, dest_path, package, build_option=None,
         except OpTestError:
             return ""
 
+    def err_message(self, er):
+        """
+        To get entire error msg
+        """
+        error_message = [str(item) for item in er]
+        combined_error = '\n'.join(error_message)
+        combined_error = re.sub(r'\x1b\[[0-9;]*[mK]|,', '', combined_error)
+        parts = combined_error.split('\x1b[')
+        cleaned_error = ''.join(char for char in combined_error if ord(char) < 128)
+        pattern = r'in file.*?compilation terminated'
+        match = re.search(pattern, cleaned_error, re.IGNORECASE | re.DOTALL)
+        if match:
+            relevant_part = match.group(0)
+            return relevant_part
+        return " "
+
+    def get_email(self, commit_id):
+        """
+        To get email of Author from the identified bad commit
+        """
+        connection = self.conf.host()
+        try :
+            connection.host_run_command("git config --global color.ui true")
+            result = connection.host_run_command("git show --format=%ce {} | sed -r 's/\x1B\[[0-9:]*[JKsu]//g'".format(commit_id))
+            email = result[0]
+            return email
+        except subprocess.CalledProcessError as e:
+            return None
```

Comment:
can we log error also which help to debug when any issue comes

Reply:
added log.info(e)